packages/app/components/Today/PopularTimes/PopularTimes.tsx

```diff
@@ -0,0 +1,212 @@
+  DownOutlined,
+  HourglassOutlined,
+  LeftOutlined,
+  RightOutlined,
+} from "@ant-design/icons";
+import { Heatmap } from "@koh/common";
+import { ParentSize } from "@visx/responsive";
+import { Dropdown, Menu } from "antd";
+import { chunk, uniq, mean } from "lodash";
+import React, { ReactElement, useState } from "react";
+import styled from "styled-components";
+import { formatWaitTime } from "../../../utils/TimeUtil";
+import { formatDateHour } from "./FormatDateHour";
+import TimeGraph from "./TimeGraph";
+
+const TitleRow = styled.div`
+  display: flex;
+  align-items: baseline;
+`;
+
+interface HeatmapProps {
+  heatmap: Heatmap;
+}
+
+const WeekdayDropdown = styled.h2`
+  display: flex;
+  align-items: center;
+  margin-left: 8px;
+  color: #1890ff;
+  cursor: pointer;
+`;
+
+const GraphWithArrow = styled.div`
+  display: flex;
+  align-items: center;
+  margin-bottom: 20px;
+`;
+
+const GraphContainer = styled.div`
+  flex-grow: 1;
+  min-width: 0; // Allow it to shrink after resizing
+`;
+
+const GraphArrowButtons = styled.div`
+  padding: 20px 5px;
+  font-size: 1.5em;
+  cursor: pointer;
+`;
+
+const DAYS_OF_WEEK = [
+  "Sunday",
+  "Monday",
+  "Tuesday",
+  "Wednesday",
+  "Thursday",
+  "Friday",
+  "Saturday",
+];
+
+function findWeekMinAndMax(days: Heatmap) {
+  let minHourInWeek = 24;
+  let maxHourInWeek = 0;
+  days.forEach((v, hour) => {
+    if (v >= 0) {
+      if (hour % 24 > maxHourInWeek) {
+        maxHourInWeek = hour % 24;
+      }
+      if (hour % 24 < minHourInWeek) {
+        minHourInWeek = hour % 24;
+      }
+    }
+  });
+  if (maxHourInWeek < minHourInWeek) {
+    return [0, 23];
+  }
+  return [minHourInWeek, maxHourInWeek];
+}
+
+const GraphNotes = styled.h4`
+  font-size: 14px;
+  color: #111;
+  padding-left: 40px;
+`;
+
+const BUSY = {
+  shortest: "the shortest",
+  shorter: "shorter than usual",
+  avg: "average",
+  longer: "longer than usual",
+  longest: "the longest",
+};
+
+// Mapping for text describing level of business, given the length of the unique wait times that week (to account for days without hours)
+const BUSY_TEXTS = {
+  1: [BUSY.avg],
+  2: [BUSY.shortest, BUSY.longest],
+  3: [BUSY.shortest, BUSY.avg, BUSY.longest],
+  4: [BUSY.shortest, BUSY.shorter, BUSY.longer, BUSY.longest],
+  5: [BUSY.shortest, BUSY.shorter, BUSY.avg, BUSY.longer, BUSY.longest],
+  6: [
+    BUSY.shortest,
+    BUSY.shorter,
+    BUSY.shorter,
+    BUSY.longer,
+    BUSY.longer,
+    BUSY.longest,
+  ],
+  7: [
+    BUSY.shortest,
+    BUSY.shorter,
+    BUSY.shorter,
+    BUSY.avg,
+    BUSY.longer,
+    BUSY.longer,
+    BUSY.longest,
+  ],
+};
+
+function generateBusyText(day: number, dailySumWaitTimes: number[]): string {
+  const dayWaitTime = dailySumWaitTimes[day];
+  const uniqSumWaitTimes = uniq(
+    dailySumWaitTimes.filter((v) => v >= 0).sort((a, b) => a - b)
+  );
+  const rank = uniqSumWaitTimes.indexOf(dayWaitTime);
+  return BUSY_TEXTS[uniqSumWaitTimes.length - 1][rank];
```

Comment:
If `uniqSumWaitTimes` has a max length of 7, it looks like the function will never access `BUSY_TEXTS` with key 7 since the `uniqSumWaitTimes.length` is being subtracted by 1.

Reply:
Nice Catch, I just fixed this